virtualcluster/pkg/util/mccontroller/mccontroller.go

```diff
@@ -114,13 +115,17 @@ type ClusterInterface interface {
 
 // NewMCController creates a new MultiClusterController.
 func NewMCController(objectType client.Object, objectListType client.ObjectList, rc reconciler.DWReconciler, opts ...OptConfig) (*MultiClusterController, error) {
-	kind := objectType.GetObjectKind().GroupVersionKind().Kind
+	kinds, _, err := scheme.Scheme.ObjectKinds(objectType)
```

Comment:
I believe you should be using this `scheme.Scheme` global variable from - https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/virtualcluster/pkg/syncer/util/scheme/scheme.go#L21 which is what other packages reference, this should allow a non-core registered type like a CRD to work.

Reply:
@christopherhein 
I believe you are right, the `scheme` from `controller-runtime` or `client-go` do not include CRD's scheme,
thanks